src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -83,13 +86,25 @@ class ToolsLocator {
                     version: version,
                     classifier: project.osdetector.classifier,
                     ext: 'exe']
-    project.logger.info("Resolving artifact: ${notation}")
     Dependency dep = project.dependencies.add(config.name, notation)
-    File file = config.fileCollection(dep).singleFile
-    if (!file.canExecute() && !file.setExecutable(true)) {
-      throw new GradleException("Cannot set ${file} as executable")
+
+    for (GenerateProtoTask protoTask in protoTasks) {
+      if (protoc.is(locator) || protoTask.plugins.findByName(locator.name) != null) {
```

Comment:
`protoTask.plugins` will fail because `protoTask.doneConfig()` has been called at this point. You can try to run the tests to see it.

I don't want to remove that preconditions check. An option is to add a `hasPlugin(String)` to `GenerateProtoTask`.